valhalla/baldr/traffictile.h

```diff
@@ -31,53 +31,31 @@ struct Speed {
   uint16_t congestion_level : 3; // some value from 0 to 7 to report back
   uint16_t is_scale : 1;         // treat speed as a floating point multiplier to edge speed
   uint16_t age : 4;              //
-  uint16_t has_incident : 1;     // TODO: reserved for whether edge has an incident or not
-
+  uint16_t spare : 1;            // TODO: reserved for later use
 #ifndef C_ONLY_INTERFACE
-  Speed() : speed_kmh{0}, congestion_level{0}, is_scale{0}, age{0}, has_incident{0} {
-  }
-  Speed(const Speed& other) = default;
-  Speed(const volatile Speed& other)
-      : speed_kmh{other.speed_kmh}, congestion_level{other.congestion_level},
-        is_scale{other.is_scale}, age{other.age}, has_incident{other.has_incident} {
+  inline bool valid() const volatile {
+    return speed_kmh > 0 || congestion_level >= 1;
   }
-#endif
-};
 
-// single incident record
-struct Incident {
-  uint64_t edge_index : 21;
-  uint64_t incident_type : 8;
-  uint64_t start_location : 10;
-  uint64_t length : 10;
-  uint64_t spare : 15;
-#ifndef C_ONLY_INTERFACE
-  Incident() : edge_index{0}, incident_type{0}, start_location{0}, length{0}, spare{0} {
+  inline bool closed() const volatile {
+    return speed_kmh == 0 && congestion_level >= 4;
   }
-  Incident(const Incident& other) = default;
-  Incident(const volatile Incident& other)
-      : edge_index{other.edge_index}, incident_type{other.incident_type},
-        start_location{other.start_location}, length{other.length}, spare{0} {
-  }
-  Incident& operator=(const Incident& other) = default;
 #endif
 };
 
 // per-speed-tile header
 struct TileHeader {
   uint64_t tile_id;
   uint32_t directed_edge_count;
-  uint32_t incident_buffer_size;
-  uint64_t active_incident_buffer : 1;
+  uint32_t spare;
```

Comment:
we should move this to the end of the header and maybe at a few more just in case future us needs some :smile:

Reply:
Done.